main.go

```diff
@@ -179,17 +196,6 @@ func readConfig() *core.Config {
 		return nil
 	}
 
-	if testAndExit {
-		coordinator := NewCoordinator()
-		if err := coordinator.Configure(config); err != nil {
-			logrus.WithError(err).Error("Configure pass failed.")
-		} else {
-			fmt.Println("Config OK.")
-		}
-		coordinator.Shutdown()
-		return nil
```

Comment:
Why not simply exit here instead of returning and all that refactoring?

Reply:
Because an exit in a readConfig() method just feels wrong. 
This PR is more a clean up instead of of refactoring in my point of view.